added test for exception raise in GROParser (per issue #597)

fiona-naughton opened the pull request:
Part of #597 ; specifically checking the GROParser raises IOError when it can't parse atom lines in a .gro file.

Changes made in this Pull Request:
 - added test .gro file with blank atom line
 - added the test for the exception

richardjgowers commented:
Looks good, just add yourself into CHANGELOG (just add a line like "Added tests for GRO format"), and AUTHORS

jbarnoud commented:
It looks good to me. Could you add a test with an ill formatted line (missing coordinate for instance)? Could you also use `git rebase` to squash your commit together?

Do not hesitate to ask if you have trouble with the rebase.

kain88-de commented:
a [good tutorial](http://rypress.com/tutorials/git/rebasing) for rebasing.

fiona-naughton commented:
Added a second test with a missing atom name.

I rebased and it all looks fine locally but it won't let be push to github - it comes up with "Updates were rejected because the tip of your current branch is behind its remote counterpart. Integrate the remote changes (e.g. 'git pull ...') before pushing again.". I tried pulling, rebasing and pushing again, but it still doesn't work. I'm not sure what I'm doing wrong?

jbarnoud commented:
If your local history looks as you want it to, you can use `git push --force`. Be careful, though, this option can lead to data loss.

fiona-naughton commented:
ah yes, that's worked now, thanks!

jbarnoud commented:
Good! You are almost there. It seems that the main repository changed since you started to work on your issue, and that the changes are not compatible with yours. It is probably not a big deal, most likely it is the CHANGELOG that got edited by someone else.

You should solve the conflict, because you are at the best place to know what version should be kept. To solve the conflicts, you should rebase against the `develop` branch of the main repository. I explained how to do that in [an other issue](https://github.com/MDAnalysis/mdanalysis/pull/779#issuecomment-198761364). You may encounter conflicts doinf the rebase, it is important to read what git tells you.

The first PR is always a pain, because you have to learn a lot of things. The next PRs are usually much easier as you know what to do.

fiona-naughton commented:
Thanks; I've rebased it again. It's currently showing no conflicts.

jbarnoud commented:
The PR is merged even though it appears not to be. I merged it and github went down exactly at the same moment (did I break github?).

See c282d663fcd0f80279bd35cd4459d58fd94eb5c8 for the merge.

@fiona-naughton Congratulation for your first merged contribution. Hopefully we will see plenty of others!
